File: openapi/code_samples/java/api_currency_list.java
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Arrays;
import java.util.List;

@RestController
@RequestMapping("/api/currency")
public class CurrencyController {

    @GetMapping("/currencies")
    public ResponseEntity<List<String>> getAllCurrencies() {
        List<String> supportedCurrencies = Arrays.asList("USD", "EUR", "GBP", "JPY");
        // Add more currencies here
        return ResponseEntity.ok(supportedCurrencies);
    }
}

File: openapi/code_samples/javascript/api_currency_list.js
const express = require('express');
const app = express();
const port = 3000;

app.get('/api/currency/currencies', (req, res) => {
    const supportedCurrencies = ["USD", "EUR", "GBP", "JPY"];
    // Add more currencies here
    res.status(200).json(supportedCurrencies);
});

app.listen(port, () => {
    console.log(`Server is listening at http://localhost:${port}`);
});
